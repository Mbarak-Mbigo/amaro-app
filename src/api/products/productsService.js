import data from './productsData';
import { sleep } from '../utils';

// get all products
const getProducts = async () => {
  await sleep();
  return data;
}

// addProduct
// editProduct
// removeProduct
// filterProducts

export {
  getProducts,
}
