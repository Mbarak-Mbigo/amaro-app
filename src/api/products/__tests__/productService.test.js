import { getProducts } from '../productsService';

describe('getProducts', () => {
  test('getProducts returns data', async () => {
    console.log('data', process.env.NODE_ENV);
    const results = await getProducts();
    expect(results).toMatchSnapshot();
  });
})
