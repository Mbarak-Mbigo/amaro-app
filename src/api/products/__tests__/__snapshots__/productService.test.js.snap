// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProducts getProducts returns data 1`] = `
Object {
  "products": Array [
    Object {
      "actual_price": "R$ 199,90",
      "code_color": "20002605_613",
      "color": "TAPEÇARIA",
      "color_slug": "tapecaria",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002605_615_catalog_1.jpg?1460136912",
      "installments": "3x R$ 66,63",
      "name": "VESTIDO TRANSPASSE BOW",
      "on_sale": false,
      "regular_price": "R$ 199,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5807_343_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5807_343_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5807_343_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5807_343_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5807_343_0_GG",
        },
      ],
      "style": "20002605",
    },
    Object {
      "actual_price": "R$ 99,90",
      "code_color": "20002570_614",
      "color": "PRETO",
      "color_slug": "preto",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002570_002_catalog_1.jpg?1459948578",
      "installments": "3x R$ 33,30",
      "name": "REGATA ALCINHA FOLK",
      "on_sale": false,
      "regular_price": "R$ 99,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "5723_40130843_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5723_40130843_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5723_40130843_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5723_40130843_0_G",
        },
        Object {
          "available": true,
          "size": "GG",
          "sku": "5723_40130843_0_GG",
        },
      ],
      "style": "20002570",
    },
    Object {
      "actual_price": "R$ 129,90",
      "code_color": "20002575_035",
      "color": "CARAMELO",
      "color_slug": "caramelo",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002575_027_catalog_1.jpg?1459537946",
      "installments": "3x R$ 43,30",
      "name": "TOP CROPPED SUEDE",
      "on_sale": false,
      "regular_price": "R$ 129,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5733_1000054_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5733_1000054_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5733_1000054_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5733_1000054_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5733_1000054_0_GG",
        },
      ],
      "style": "20002575",
    },
    Object {
      "actual_price": "R$ 149,90",
      "code_color": "20002581_614",
      "color": "MINI FOLK",
      "color_slug": "mini-folk",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002581_614_catalog_1.jpg?1459536611",
      "installments": "3x R$ 49,97",
      "name": "BATA DECOTE FLUID",
      "on_sale": false,
      "regular_price": "R$ 149,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5749_341_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5749_341_0_P",
        },
        Object {
          "available": false,
          "size": "M",
          "sku": "5749_341_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5749_341_0_G",
        },
        Object {
          "available": true,
          "size": "GG",
          "sku": "5749_341_0_GG",
        },
      ],
      "style": "20002581",
    },
    Object {
      "actual_price": "R$ 119,90",
      "code_color": "20002602_027",
      "color": "MARINHO",
      "color_slug": "marinho",
      "discount_percentage": "12%",
      "image": "",
      "installments": "3x R$ 39,97",
      "name": "T-SHIRT LEATHER DULL",
      "on_sale": true,
      "regular_price": "R$ 139,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "5793_1000032_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5793_1000032_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5793_1000032_0_M",
        },
        Object {
          "available": false,
          "size": "G",
          "sku": "5793_1000032_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5793_1000032_0_GG",
        },
      ],
      "style": "20002602",
    },
    Object {
      "actual_price": "R$ 189,90",
      "code_color": "20002584_027",
      "color": "BURGUNDY",
      "color_slug": "burgundy",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002584_035_catalog_1.jpg?1459947139",
      "installments": "3x R$ 63,30",
      "name": "CAMISA SUEDE SPAN",
      "on_sale": false,
      "regular_price": "R$ 189,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5755_1000070_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5755_1000070_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5755_1000070_0_M",
        },
        Object {
          "available": false,
          "size": "G",
          "sku": "5755_1000070_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5755_1000070_0_GG",
        },
      ],
      "style": "20002584",
    },
    Object {
      "actual_price": "R$ 159,90",
      "code_color": "20002634_613",
      "color": "FREE DOTS",
      "color_slug": "free-dots",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002634_613_catalog_1.jpg?1459548109",
      "installments": "3x R$ 53,30",
      "name": "CALÇA CLASSIC PRINT",
      "on_sale": false,
      "regular_price": "R$ 159,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "36",
          "sku": "5865_339_0_36",
        },
        Object {
          "available": true,
          "size": "38",
          "sku": "5865_339_0_38",
        },
        Object {
          "available": true,
          "size": "40",
          "sku": "5865_339_0_40",
        },
        Object {
          "available": true,
          "size": "42",
          "sku": "5865_339_0_42",
        },
        Object {
          "available": true,
          "size": "44",
          "sku": "5865_339_0_44",
        },
      ],
      "style": "20002634",
    },
    Object {
      "actual_price": "R$ 99,90",
      "code_color": "20002570_612",
      "color": "OFF WHITE",
      "color_slug": "off-white",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002570_029_catalog_1.jpg?1459948578",
      "installments": "3x R$ 33,30",
      "name": "REGATA ALCINHA FOLK",
      "on_sale": false,
      "regular_price": "R$ 99,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "5723_1000058_0_PP",
        },
        Object {
          "available": false,
          "size": "P",
          "sku": "5723_1000058_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5723_1000058_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5723_1000058_0_G",
        },
        Object {
          "available": true,
          "size": "GG",
          "sku": "5723_1000058_0_GG",
        },
      ],
      "style": "20002570",
    },
    Object {
      "actual_price": "R$ 149,90",
      "code_color": "20002612_612",
      "color": "FLORAL RED",
      "color_slug": "floral-red",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002612_612_catalog_1.jpg?1459531023",
      "installments": "3x R$ 49,97",
      "name": "MACAQUINHO ALFAIATARIA PEACE",
      "on_sale": false,
      "regular_price": "R$ 149,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5819_337_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5819_337_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5819_337_0_M",
        },
        Object {
          "available": false,
          "size": "G",
          "sku": "5819_337_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5819_337_0_GG",
        },
      ],
      "style": "20002612",
    },
    Object {
      "actual_price": "R$ 159,90",
      "code_color": "20002945_027",
      "color": "CARAMELO",
      "color_slug": "caramelo",
      "discount_percentage": "25%",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002945_027_catalog_1.jpg?1459540966",
      "installments": "3x R$ 53,30",
      "name": "BOLSA FLAP TRIANGLE",
      "on_sale": true,
      "regular_price": "R$ 199,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "U",
          "sku": "6559_1000054_0_U",
        },
      ],
      "style": "20002945",
    },
    Object {
      "actual_price": "R$ 219,90",
      "code_color": "20002606_612",
      "color": "TAPEÇARIA",
      "color_slug": "tapecaria",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002606_615_catalog_1.jpg?1459515279",
      "installments": "3x R$ 73,30",
      "name": "VESTIDO LONGO FLOAT",
      "on_sale": false,
      "regular_price": "R$ 219,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "5811_343_0_PP",
        },
        Object {
          "available": false,
          "size": "P",
          "sku": "5811_343_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5811_343_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5811_343_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5811_343_0_GG",
        },
      ],
      "style": "20002606",
    },
    Object {
      "actual_price": "R$ 149,90",
      "code_color": "20002581_612",
      "color": "BOHEMIAN PRETO",
      "color_slug": "bohemian-preto",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002581_611_catalog_1.jpg?1459536611",
      "installments": "3x R$ 49,97",
      "name": "BATA DECOTE FLUID",
      "on_sale": false,
      "regular_price": "R$ 149,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5749_333_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5749_333_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5749_333_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5749_333_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5749_333_0_GG",
        },
      ],
      "style": "20002581",
    },
    Object {
      "actual_price": "R$ 69,90",
      "code_color": "20001609_029",
      "color": "PRETO",
      "color_slug": "preto",
      "discount_percentage": "50%",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20001609_002_catalog_1.jpg?",
      "installments": "2x R$ 34,95",
      "name": "VESTIDO FRANZIDO RECORTES",
      "on_sale": true,
      "regular_price": "R$ 139,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "3627_40130843_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "3627_40130843_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "3627_40130843_0_M",
        },
        Object {
          "available": false,
          "size": "G",
          "sku": "3627_40130843_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "3627_40130843_0_GG",
        },
      ],
      "style": "20001609",
    },
    Object {
      "actual_price": "R$ 84,90",
      "code_color": "20001786_594",
      "color": "PAISLEY GIPSY AZUL",
      "color_slug": "paisley-gipsy-azul",
      "discount_percentage": "39%",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20001786_594_catalog_1.jpg?1449159646",
      "installments": "2x R$ 42,45",
      "name": "CALÇA COMFORT TASSEL",
      "on_sale": true,
      "regular_price": "R$ 139,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "36",
          "sku": "4029_259_0_36",
        },
        Object {
          "available": true,
          "size": "38",
          "sku": "4029_259_0_38",
        },
        Object {
          "available": true,
          "size": "40",
          "sku": "4029_259_0_40",
        },
        Object {
          "available": true,
          "size": "42",
          "sku": "4029_259_0_42",
        },
        Object {
          "available": false,
          "size": "44",
          "sku": "4029_259_0_44",
        },
      ],
      "style": "20001786",
    },
    Object {
      "actual_price": "R$ 69,90",
      "code_color": "20001695_063",
      "color": "INDIGO ESCURO",
      "color_slug": "indigo-escuro",
      "discount_percentage": "50%",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20001695_063_catalog_1.jpg?",
      "installments": "2x R$ 34,95",
      "name": "SAIA DENIM BOTÕES",
      "on_sale": true,
      "regular_price": "R$ 139,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "36",
          "sku": "3787_233_0_36",
        },
        Object {
          "available": true,
          "size": "38",
          "sku": "3787_233_0_38",
        },
        Object {
          "available": true,
          "size": "40",
          "sku": "3787_233_0_40",
        },
        Object {
          "available": true,
          "size": "42",
          "sku": "3787_233_0_42",
        },
        Object {
          "available": false,
          "size": "44",
          "sku": "3787_233_0_44",
        },
      ],
      "style": "20001695",
    },
    Object {
      "actual_price": "R$ 149,90",
      "code_color": "20002600_293",
      "color": "LARANJA QUEIMADO",
      "color_slug": "laranja-queimado",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002600_293_catalog_1.jpg?1460416037",
      "installments": "3x R$ 49,97",
      "name": "BLUSA LAÇO ISTAMBUL",
      "on_sale": false,
      "regular_price": "R$ 149,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "PP",
          "sku": "5789_311_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5789_311_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5789_311_0_M",
        },
        Object {
          "available": false,
          "size": "G",
          "sku": "5789_311_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5789_311_0_GG",
        },
      ],
      "style": "20002600",
    },
    Object {
      "actual_price": "R$ 149,90",
      "code_color": "20002829_006",
      "color": "BEGE",
      "color_slug": "bege",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002829_006_catalog_1.jpg?1459550820",
      "installments": "3x R$ 49,97",
      "name": "BLUSA RIB STRIPES",
      "on_sale": false,
      "regular_price": "R$ 149,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "6299_1000012_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "6299_1000012_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "6299_1000012_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "6299_1000012_0_G",
        },
        Object {
          "available": true,
          "size": "GG",
          "sku": "6299_1000012_0_GG",
        },
      ],
      "style": "20002829",
    },
    Object {
      "actual_price": "R$ 239,90",
      "code_color": "20002648_029",
      "color": "OFF-WHITE",
      "color_slug": "off-white",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20002648_029_catalog_1.jpg?1459552552",
      "installments": "3x R$ 79,97",
      "name": "CASACO WHITE FUR",
      "on_sale": false,
      "regular_price": "R$ 239,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "5891_1000058_0_PP",
        },
        Object {
          "available": true,
          "size": "P",
          "sku": "5891_1000058_0_P",
        },
        Object {
          "available": true,
          "size": "M",
          "sku": "5891_1000058_0_M",
        },
        Object {
          "available": true,
          "size": "G",
          "sku": "5891_1000058_0_G",
        },
        Object {
          "available": false,
          "size": "GG",
          "sku": "5891_1000058_0_GG",
        },
      ],
      "style": "20002648",
    },
    Object {
      "actual_price": "R$ 59,90",
      "code_color": "20001684_587",
      "color": "FLORAL VINTAGE VERMELHO",
      "color_slug": "floral-vintage-vermelho",
      "discount_percentage": "54%",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20001684_585_catalog_1.jpg?",
      "installments": "1x R$ 59,90",
      "name": "VESTIDO CURTO FESTIVAL",
      "on_sale": true,
      "regular_price": "R$ 129,90",
      "sizes": Array [
        Object {
          "available": false,
          "size": "PP",
          "sku": "3765_211_0_PP",
        },
        Object {
          "available": false,
          "size": "P",
          "sku": "3765_211_0_P",
        },
        Object {
          "available": false,
          "size": "M",
          "sku": "3765_211_0_M",
        },
        Object {
          "available": false,
          "size": "G",
          "sku": "3765_211_0_G",
        },
        Object {
          "available": true,
          "size": "GG",
          "sku": "3765_211_0_GG",
        },
      ],
      "style": "20001684",
    },
    Object {
      "actual_price": "R$ 49,90",
      "code_color": "20001882_008",
      "color": "MARRROM",
      "color_slug": "marrom",
      "discount_percentage": "55%",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20001882_008_catalog_1.jpg?",
      "installments": "1x R$ 49,90",
      "name": "ÓCULOS DE SOL BOLD",
      "on_sale": true,
      "regular_price": "R$ 109,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "U",
          "sku": "4229_1000016_0_U",
        },
      ],
      "style": "20001882",
    },
    Object {
      "actual_price": "R$ 49,90",
      "code_color": "20001883_019",
      "color": "CINZA",
      "color_slug": "cinza",
      "discount_percentage": "55%",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20001883_019_catalog_1.jpg?",
      "installments": "1x R$ 49,90",
      "name": "ÓCULOS DE SOL AVIADOR VINTAGE",
      "on_sale": true,
      "regular_price": "R$ 109,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "U",
          "sku": "4231_1000038_0_U",
        },
      ],
      "style": "20001883",
    },
    Object {
      "actual_price": "R$ 29,90",
      "code_color": "20001913_009",
      "color": "dourado",
      "color_slug": "dourado",
      "discount_percentage": "",
      "image": "https://d3l7rqep7l31az.cloudfront.net/images/products/20001913_009_catalog_1.jpg?",
      "installments": "1x R$ 29,90",
      "name": "PULSEIRA STYLISH",
      "on_sale": false,
      "regular_price": "R$ 29,90",
      "sizes": Array [
        Object {
          "available": true,
          "size": "U",
          "sku": "4279_1000018_0_U",
        },
      ],
      "style": "20001913",
    },
  ],
}
`;
